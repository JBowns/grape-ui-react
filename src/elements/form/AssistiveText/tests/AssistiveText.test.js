import React from 'react';
import { AssistiveText } from '../';
import 'jest-styled-components';
import renderer from 'react-test-renderer';

const assertReactElement = element => {
  const component = renderer.create(element);
  return component.toJSON();
};
describe('Assistive Text Component base', () => {
  it('should return an Assistive Text object', () => {
    const element = <AssistiveText id="assitiveTextIdHelp">Helper text is here</AssistiveText>;
    expect(assertReactElement(element)).toMatchSnapshot();
  });
  it('should return an Assistive Text object with danger text color', () => {
    const element = <AssistiveText color="brandDanger" id="assitiveTextIdError">Error text is here</AssistiveText>;
    expect(assertReactElement(element)).toMatchSnapshot();
  });
});
