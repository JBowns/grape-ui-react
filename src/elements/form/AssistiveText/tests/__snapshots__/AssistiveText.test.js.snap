// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Assistive Text Component base should return an Assistive Text object 1`] = `
.c0 {
  color: hsl(0,0%,50%);
  font-family: inherit;
  font-size: 0.8rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  padding: 0 1rem;
}

<div
  className="c0"
  id="assitiveTextIdHelp"
>
  Helper text is here
</div>
`;

exports[`Assistive Text Component base should return an Assistive Text object with danger text color 1`] = `
.c0 {
  color: hsl(4.1,89.6%,58.4%);
  font-family: inherit;
  font-size: 0.8rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  padding: 0 1rem;
}

<div
  className="c0"
  id="assitiveTextIdError"
>
  Error text is here
</div>
`;
